pywinauto/application.py

```diff
@@ -839,6 +840,15 @@ def __connect(self, **kwargs):
     def connect(self, **kwargs):
         """Connects to an already running process"""
         connected = False
+
+        if 'timeout' in kwargs:
+            try:
+                self.timeout = float(kwargs['timeout'])
+                time.sleep(self.timeout)
+            except Exception as e:
+                message = "timeout isn't correct"
+                raise AppNotConnected(str(e) + message)

```

Comment:
The exception should be ```ValueError``` for the wrong argument. Also using ```time.sleep``` is a naive approach. The method should try to check it several times during the timeout period and continue as soon as the app successfully connected. You might try to use ```WaitUntilPasses``` with a list of possible exceptions for that purpose or even
reuse an idea with a decorator that I suggested in another PR: https://github.com/pywinauto/pywinauto/pull/193#discussion_r62149644